### ISSUE TITLE
shards install : crystal/shards not found 

### ISSUE THREAD

**Issue by davlgd** [crystal-lang/crystal#13889]
When I execute `shards install`, even on a just init empty app, I get this error : 

```
Resolving dependencies
Writing shard.lock
Could not execute 'crystal': Error executing process: 'crystal': No such file or directory
```

On some projects, I have the same message, with `shards` missing. 

I think the problem is that on the VM I use, `crystal`/`shards` are in the `$HOME/.local/bin` folder (it's in `$PATH`). I can't change that. 

My `crystal` version :

```
Crystal 1.10.1 [c6f3552f5] (2023-10-13)

LLVM: 15.0.7
Default target: x86_64-unknown-linux-gnu
```

**Comment by lasso** [comment]
Does the shell find the correct executables?

What does `which crystal` and `which shards` tell you?

**Comment by davlgd** [comment]
Yes, I use `crystal` as a command in a build script and everything is fine. But not with `shards install`. 

>>>
~ $ which crystal 
/home/bas/.local/bin/crystal
~ $ which shards
/home/bas/.local/bin/shards
>>>

**Comment by davlgd** [comment]
For example : 

>>>
$ shards init
Created shard.yml
$ shards install
Resolving dependencies
Writing shard.lock
Could not execute 'crystal': Error executing process: 'crystal': No such file or directory
$ crystal
Usage: crystal [command] [switches] [program file] [--] [arguments]

Command:
    init                     generate a new project
    build                    build an executable
    clear_cache              clear the compiler cache
    docs                     generate documentation
    env                      print Crystal environment information
    eval                     eval code from args or standard input
    i/interactive            starts interactive Crystal
    play                     starts Crystal playground server
    run (default)            build and run program
    spec                     build and run specs (in spec directory)
    tool                     run a tool
    help, --help, -h         show this help
    version, --version, -v   show version

Run a command followed by --help to see command specific information, ex:
    crystal <command> --help

>>>

**Comment by straight-shoota** [comment]
What happens if you run `CRYSTAL=/home/bas/.local/bin/crystal shards install`?

**Comment by lasso** [comment]
You seem to be hitting https://github.com/crystal-lang/shards/blob/master/src/config.cr#L88 and that seems affected by the environment variable CRYSTAL_VERSION. Does `env | grep CRYSTAL_VERSION` return anything?

**Comment by straight-shoota** [comment]
@lasso If `CRYSTAL_VERSION` was set, shards would not try to execute the `crystal` command. The only purpose for that is to determine the version.

**Comment by davlgd** [comment]
> What happens if you run `CRYSTAL=/home/bas/.local/bin/crystal shards install`?

it's fine

> You seem to be hitting https://github.com/crystal-lang/shards/blob/master/src/config.cr#L88 and that seems affected by the environment variable CRYSTAL_VERSION. Does `env | grep CRYSTAL_VERSION` return anything?

it's empty

**Comment by straight-shoota** [comment]
If `CRYSTAL=/home/bas/.local/bin/crystal shards install` works, it seems like an issue with your `PATH` configuration. Maybe it's not properly forwarded to `shards`...?
I don't think `shards` can do anything about this.

**Comment by davlgd** [comment]
If it was not properly forwarded, `shards` or `crystal` command wouldn't work. I get a "No such file or directory" only when `shards` is launching `crystal`

**Comment by straight-shoota** [comment]
Let's check that with a simple script that prints $PATH to stderr and fails:
```sh
echo 'echo $PATH >&2; exit 1' > test.sh
chmod +x test.sh
CRYSTAL=$PWD/test.sh shards install
```
Does the output contain `/home/bas/.local/bin/`?

And what's the output of `crystal eval 'puts Process.find_executable("crystal")'`?
